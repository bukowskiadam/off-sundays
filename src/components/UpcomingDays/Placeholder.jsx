import React from 'react';

import style from './Placeholder.module.css';

const Placeholder = () => (
    <div className={ style.container } />
);

export default Placeholder;
