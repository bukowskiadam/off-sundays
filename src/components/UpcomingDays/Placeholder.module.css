@keyframes blink {
    from {
        opacity: 0;
    }
    to {
        opacity: 1;
    }
}

.container::before {
    content: 'ustalanie daty...';
    margin: 5rem auto;
    display: block;
    animation: blink 1s linear infinite alternate;
}
