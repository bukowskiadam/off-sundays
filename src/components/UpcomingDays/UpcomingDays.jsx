import React from 'react';
import PropTypes from 'prop-types';

import { sunday, nextWeek } from '../../helpers/dateOperations';
import { isClosed } from '../../data';

import Day from '../Day';
import DayMini from '../DayMini';
import BoxContainer from '../BoxContainer';

const UpcomingDays = ({ upcomingSunday }) => {
    const nextSunday = nextWeek(upcomingSunday);
    const twoWeeksLater = nextWeek(nextSunday);
    const threeWeeksLater = nextWeek(twoWeeksLater);

    return (
        <BoxContainer>
            <Day date={ upcomingSunday } isClosed={ isClosed(upcomingSunday) } />
            <DayMini date={ nextSunday } isClosed={ isClosed(nextSunday) } />
            <DayMini date={ twoWeeksLater } isClosed={ isClosed(twoWeeksLater) } />
            <DayMini date={ threeWeeksLater } isClosed={ isClosed(threeWeeksLater) } />
        </BoxContainer>
    );
};

UpcomingDays.propTypes = {
    upcomingSunday: PropTypes.instanceOf(Date),
};

UpcomingDays.defaultProps = {
    upcomingSunday: sunday(),
};

export default UpcomingDays;
