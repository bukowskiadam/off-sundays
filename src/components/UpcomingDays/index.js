import forClientOnly from '../../helpers/forClientOnly';
import UpcomingDays from './UpcomingDays';
import Placeholder from './Placeholder';

export default forClientOnly(UpcomingDays, Placeholder);
