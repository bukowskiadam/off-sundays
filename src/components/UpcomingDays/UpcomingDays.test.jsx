import React from 'react';
import { shallow } from 'enzyme';

import UpcomingDays from './UpcomingDays';

describe('UpcomingDays', () => {
    it('renders correctly', () => {
        const sunday = new Date('2018-03-18');
        const wrapper = shallow(<UpcomingDays upcomingSunday={sunday} />);

        expect(wrapper).toMatchSnapshot();
    });
});
