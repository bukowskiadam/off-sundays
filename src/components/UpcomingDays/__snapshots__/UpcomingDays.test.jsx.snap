// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpcomingDays renders correctly 1`] = `
<BoxContainer>
  <Day
    date={2018-03-18T00:00:00.000Z}
    isClosed={true}
  />
  <Day
    date={2018-03-25T00:00:00.000Z}
    isClosed={false}
  />
  <Day
    date={2018-03-31T23:00:00.000Z}
    isClosed={true}
  />
  <Day
    date={2018-04-07T23:00:00.000Z}
    isClosed={true}
  />
</BoxContainer>
`;
